internal/controller/container_reconciler.go

```diff
@@ -58,6 +63,40 @@ func (r *ContainerImageReconciler) ReconcileContainerImage(ctx context.Context,
 		return result, err
 	}
 
+	if r.Kind == "Notebook" {
+		nb := obj.(*apiv1.Notebook)
+
+		// this is a signed url request for a file with a different md5 checksum
+		if nb.Spec.Image.Upload.Md5Checksum != nb.Status.LastGeneratedMd5Checksum {
+			url, err := r.callSignedUrlGenerator(nb, r.CloudManagerGrpcConn)
+			if err != nil {
+				return result{}, fmt.Errorf("generating upload url: %w", err)
+			}
+
+			nb.Status.UploadURL = url
+			nb.Status.LastGeneratedMd5Checksum = nb.Spec.Image.Upload.Md5Checksum
+			// TODO(bjb): shouldn't this happen in the notebook_controller?
+			// The reconciler here doesn't have Status()
+			// if err := r.Status().Update(ctx, &nb); err != nil {
+			// 	return result{}, fmt.Errorf("updating notebook status: %w", err)
+			// }
```

Comment:
@nstogner @samos123 - I tested this out and confirmed when I run the controller I get hung up here. Guessing Nick has a way to do this.

```
2023-07-17T12:15:06-07:00       INFO    Done reconciling Notebook       {"controller": "notebook", "controllerGroup": "substratus.ai", "controllerKind": "Notebook", "Notebook": {"name":"falcon-7b-instruct","namespace":"default"}, "namespace": "default", "name": "falcon-7b-instruct", "reconcileID": "f7f1f2d1-52f0-4adb-b369-c1c974eada8e"}
2023-07-17T12:15:06-07:00       ERROR   Reconciler error        {"controller": "notebook", "controllerGroup": "substratus.ai", "controllerKind": "Notebook", "Notebook": {"name":"falcon-7b-instruct","namespace":"default"}, "namespace": "default", "name": "falcon-7b-instruct", "reconcileID": "f7f1f2d1-52f0-4adb-b369-c1c974eada8e", "error": "updating notebook status: Operation cannot be fulfilled on notebooks.substratus.ai \"falcon-7b-instruct\": the object has been modified; please apply your changes to the latest version and try again"}
```